### PR TITLE
Add a download page

### PR DESCRIPTION
**Pull request by afshin**
This PR adds a new page to the site for downloading the miniforge installer for a given OS/arch.

Here is what it looks like:

### Light
Screenshot 2024-03-12 at 11 10 36

### Dark
Screenshot 2024-03-12 at 11 10 50

### CODE REVIEW

**Comment by beckermr** [comment]
The rendering on mobile has the platform logos not inside the grey boxes.

File: src/components/Header/index.jsx
Comment (jaimergp) [inline src/components/Header/index.jsx:33]:
```suggestion
                        Download Miniforge

```

File: src/components/Header/index.jsx
Comment (jaimergp) [inline src/components/Header/index.jsx:31]: What if we use `/miniforge/` here? I assume this is superseding https://conda-forge.org/miniforge

File: src/pages/download.jsx
Comment (jaimergp) [inline src/pages/download.jsx:17]:
```suggestion
                    Download Miniforge
```

File: docusaurus.config.js
Comment (jaimergp) [inline docusaurus.config.js:351]:
```suggestion
            href: "/miniforge",
            label: "Miniforge",
```

File: src/pages/download.jsx
Comment (SylvainCorlay) [reply src/pages/download.jsx:17]: There will probably be a lot of bikeshedding about the formulation on this page (what more explanatory text to include). I suggest we get the feature in, and let the core devs make such edits after.

File: src/components/Download/index.jsx
Comment (jaimergp) [inline src/components/Download/index.jsx:75]: Since they are monochrome, would it be better to use `filter: invert(1)` CSS here and save one image?

File: src/components/Download/index.jsx
Comment (jaimergp) [inline src/components/Download/index.jsx:71]:
```suggestion
                      title={`Download Miniforge installer for ${os} ${arch}`}
```

File: src/components/Header/index.jsx
Comment (jaimergp) [reply src/components/Header/index.jsx:31]: If I read "Download" on a website for a repository of packages, I don't know if I'd expect to find an installer for _one_ of the package managers that are able to consume those packages. So if we want to make this a Download page, it should inform about Miniforge, micromamba, and pixi (so far).

File: src/pages/download.jsx
Comment (SylvainCorlay) [reply src/pages/download.jsx:17]: Although, 2c: "Conda-forge installer" is a lot easier to understand to a newcomer on this page than "Miniforge".

File: src/components/Header/index.jsx
Comment (SylvainCorlay) [reply src/components/Header/index.jsx:31]: The download page is where you find stuff to download. There could be more things than the installer eventually.

Also, "Miniforge" as a topbar item is not something someone new to the project will understand.

**Comment by afshin** [comment]
From a branding point of view, I would rename `Miniforge` to `mini-forge` so that it looks more like `conda-forge`.

But of course, I'll leave it to you all to decide what the language and terminology you want on this page is.

I will say that as an end user who has heard of `conda-forge` and has landed on `conda-forge.org`, I would be confused what `Miniforge` is, perhaps.

File: src/components/Download/index.jsx
Comment (afshin) [reply src/components/Download/index.jsx:75]: Clever!

**Comment by beckermr** [comment]
A little explanation of the thumbs down might help clarify. 

So miniforge is reference to miniconda which is a similar thing provided by others for a barebones conda install. We should not add the dash for this reason.

**Comment by afshin** [comment]
> A little explanation of the thumbs down might help clarify.
> 
> So miniforge is reference to miniconda which is a similar thing provided by others for a barebones conda install. We should not add the dash for this reason.

Thanks! This is helpful to know.

**Comment by h-vetinari** [comment]
I think the "mini" is obsolete. It's "mini" only relative to the thankfully long-gone practice of providing a "full" distribution of packages as the baseline (think numpy, pandas, matplotlib, etc.), whereas the "mini" was just what you needed to run conda itself (and then create your own environments).

No-one uses conda-forge this way; we don't have a "full" distribution that people can install. So I completely agree with @SylvainCorlay: 
> "Conda-forge installer" is a lot easier to understand to a newcomer on this page than "Miniforge".

**Comment by jakirkham** [comment]
If we want to have a discussion about naming, we should do that in a separate issue dedicated to that

Let's focus on the page proposed in this PR

**Comment by SylvainCorlay** [comment]
I would be in favor of getting this in - as it is an improvement to the current state of things, providing an easier way to get started. More prose can easily be added by the core team.

**Comment by isuruf** [comment]
I think it's okay to have this merged as-is.

I'm not in favour of renaming Miniforge to mini-forge or anything else. It's a brand by itself now and changing it is bad.

**Review by SylvainCorlay — APPROVED** [review]

**Review by jaimergp — APPROVED** [review]
I think this would benefit from some intro text explaining what they are downloading and how to run it (or at least link to the official instructions somewhere). A link to the Miniforge Releases or the conda-forge.org/miniforge website could also be nice in case people want to check the sha256s or so.

But we can amend those later, let's get this in.

**Comment by beckermr** [comment]
We never changed to Miniforge. I think Isuru thumbs upped on that condition?

**Comment by beckermr** [comment]
Or maybe I was confused actually.

**Comment by afshin** [comment]
I will be away on vacation so please don't hesitate to push to this branch any changes that are necessary. Thank you!

**Comment by jakirkham** [comment]
Thanks Afshin! 🙏

Have a good vacation 🏖️

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Should we merge?

**Comment by beckermr** [comment]
Yes we should.

**Comment by jakirkham** [comment]
Thanks all! 🙏

If anything needs follow up, lets submit PRs to tweak it

**Pull request merged**
